pkg/core/dao/dao.go

```diff
@@ -672,6 +673,44 @@ func (dao *Simple) StoreAsBlock(block *block.Block, buf *io.BufBinWriter) error
 	return dao.Store.Put(key, buf.Bytes())
 }
 
+// DeleteBlock removes block from dao.
+func (dao *Simple) DeleteBlock(h util.Uint256) error {
+	return DeleteBlocks(dao.Store, dao.network, dao.stateRootInHeader, h)
+}
+
+// DeleteBlocks removes block from provided storage.
+func DeleteBlocks(s storage.Store, network netmode.Magic, stateRootInHeader bool, hs ...util.Uint256) error {
+	batch := s.Batch()
+	key := make([]byte, util.Uint256Size+1)
+	for i := range hs {
+		key[0] = byte(storage.DataBlock)
+		copy(key[1:], hs[i].BytesLE())
+		bs, err := s.Get(key)
+		if err != nil {
+			return err
+		}
+
+		b, err := block.NewBlockFromTrimmedBytes(network, stateRootInHeader, bs)
+		if err != nil {
+			return err
+		}
+
+		key[0] = byte(storage.DataTransaction)
+		for _, tx := range b.Transactions {
+			copy(key[1:], tx.Hash().BytesLE())
+			batch.Delete(key)
+			key[0] = byte(storage.STNotification)
+			copy(key[1:], tx.Hash().BytesBE())
+			batch.Delete(key)
+		}
+
```

Comment:
This leaves block data as is (with hashes and consensus data), but we actually want to only have headers there.

Reply:
Fixed.